Modules/Challenges/test/Challenges.API.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -0,0 +1,157 @@
+using Challenges.API.Tests.Integration.API;
+using Challenges.API.Tests.Integration.Extensions;
+using Challenges.API.Tests.Integration.Models;
+using Microsoft.Extensions.Options;
+using TechTalk.SpecFlow.Assist;
+using static Challenges.API.Tests.Integration.Configuration.Settings;
+
+namespace Challenges.API.Tests.Integration.StepDefinitions;
+
+[Binding]
+public class ChallengesApiStepDefinitions
+{
+    private readonly ChallengesApi _challengeApi;
+    private string _challengeId;
+    private HttpResponse<ChallengeResponse> _challengeResponse;
+    private readonly RequestConfiguration _requestConfiguration;
+
+    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, ChallengesApi challengeApi)
+    {
+        _challengeApi = challengeApi;
+        _challengeId = string.Empty;
+        _challengeResponse = new HttpResponse<ChallengeResponse>();
+        _requestConfiguration = new RequestConfiguration
+        {
+            AuthenticationParameters = new AuthenticationParameters
+            {
+                GrantType = "password",
+                ClientId = httpConfiguration.Value.ClientCredentials.ClientId,
+                ClientSecret = httpConfiguration.Value.ClientCredentials.ClientSecret,
+                UserName = "USRa",
+                Password = "a"
+            }
+        };
+    }
+
+    [Given(@"the user is authenticated")]
+    public void GivenTheUserIsAuthenticated()
+    {
+        _requestConfiguration.IsAuthenticated = true;
+    }
+
+    [Given(@"the user is unauthenticated")]
+    public void GivenTheUserIsUnauthenticated()
+    {
+        _requestConfiguration.IsAuthenticated = false;
+    }
+
+    [Given(@"the Accept header is '([^']*)'")]
+    public void GivenTheAcceptHeaderIs(string acceptHeader)
+    {
+        _requestConfiguration.AcceptHeader = acceptHeader;
+    }
+
+    [Given(@"a Challenge c")]
+    public async Task GivenAChallengeC()
+    {
+        var challengeResponse = await _challengeApi.CreateChallenge(_requestConfiguration);
+        challengeResponse.IsSuccessStatusCode.Should().BeTrue();
+
+        _challengeId = challengeResponse.Data!.Result!.Id;
+        _challengeId.Should().NotBeNullOrEmpty(because: "Required value for 'Id' is missing.");
+    }
+
+    [When(@"a POST request is sent to the Challenges endpoint with")]
+    public async Task WhenAPOSTRequestIsSentToTheChallengesEndpointWith(Table table)
+    {
+        var requestConfiguration = table.CreateInstance<RequestConfiguration>();
```

Comment:
Doesn't the requestConfiguration from the table have to be merged with `_requestConfiguration`?

Reply:
Given the RequestConfiguration class:
```c#
public class RequestConfiguration
{
    public AuthenticationParameters AuthenticationParameters { get; set; } = new AuthenticationParameters();
    public bool Authenticate { get; set; } = false;
    public string ContentType { get; set; } = string.Empty;
    public string AcceptHeader { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
}
```
If the authentication is not relevant for the test (in this test we don't specify if the user is anonymous or authenticated), we don't really need to merge requestConfiguration from the table with _requestConfiguration. All the relevant properties are filled in the scenario table.

Reply:
But you don't know what other properties of the RequestConfiguration defaults are created for in the future. So I think it would be better/safer to just merge them